### DISCUSSION TITLE
Tailwind ships tons of unused boilerplate CSS

### DISCUSSION THREAD

**Discussion by l-Fingon-l** [tailwindlabs/tailwindcss#7538] [Help] [answered]
Is it just me or does Tailwind in fact ship a lot of useless CSS?
--
The [main page](https://tailwindcss.com/) clearly states:
![image](https://user-images.githubusercontent.com/70348701/154845410-7468a7d9-278e-4675-8ae1-f58819967e2e.png)
> Tailwind automatically removes all unused CSS when building for production  

However, it does not seem to be the case for me. I have just literally tried copying the example above but ended up with a huge pile of unpurged and basically unused utility boilerplate CSS:  
![image](https://user-images.githubusercontent.com/70348701/154845741-c60312f3-c6b6-404e-814b-5c3b80404d13.png)

8KB: 
![image](https://user-images.githubusercontent.com/70348701/154845860-93a608a3-4fad-476e-9138-4de62750bca6.png)

The `--minify` option just eliminates the spaces but all the unnecessary CSS is still there.
---
### Is it just me doing something wrong or is it a bug? Would love to hear any ideas from you

**Comment by adamwathan** [comment, marked as answer]
Our [Preflight styles](https://tailwindcss.com/docs/preflight) (the base reset) is always included at the moment. We may explore trying to detect what parts you don't need based on your `content` config and strip those parts out in the future though.

There isn't any unused utility CSS in there though, everything aside from the browser normalization base rules is generated on demand for your specific content.

I think "tons of unused boilerplate" is a bit of an exaggeration — I can understand why that would be your gut reaction but the whole base layer is actually only about 1kb minified and compressed, and most of the rules in there are going to be needed for every project (stuff targeting `html`, `body`, `a`, etc). Stripping out some of the more obscure stuff you might not actually need like `sub`, `sup`, `abbr`, and a couple of others isn't going to change anything about the characteristics of your bundle or the performance of your site, which is why we haven't invested time into trying to strip that stuff out when it's not needed. Hope that makes sense.

**Reply by l-Fingon-l** [reply]
Thank you so much for the reply! 
Yes, it makes a lot of sense. The only reason I used the "tons of unused boilerplate" phrase was just to make an emphasis on what was the main part of the question. That's certainly not much and that is certainly something any project would need realistically.

Just was curious about the nature of this CSS in the output and whether I was doing something wrong.
Once again, thanks for your answer, it makes everything crystal-clear now <3
